api-aws-lambda/src/main/kotlin/org/coepi/api/v4/TCNCloudAPIHandler.kt

```diff
@@ -125,13 +135,9 @@ class TCNCloudAPIHandler: RequestHandler<APIGatewayProxyRequestEvent, APIGateway
     fun handlePostReport(input: APIGatewayProxyRequestEvent): APIGatewayProxyResponseEvent {
         return try {
             val reportData = Base64.getDecoder().decode(input.body)
-            val timestamp = Instant.now().toEpochMilli()
-            val date = LocalDate.now(ZoneId.of("UTC"))
-            val intervalNumber = generateIntervalForTimestamp(timestamp)
 
-            // TODO: Validate reportData and signature
-            reportsDao.addReport(reportData, date, intervalNumber, timestamp)
-            logger.info("Successfully added report with intervalNumber: $intervalNumber and date: $date")
+            val savedReport = reportService.saveReport(reportData)
+            logger.info("Successfully added report ${savedReport.reportId}")
 
             APIGatewayProxyResponseEvent()
```

Comment:
This logic can also be moved to the core. The core service can return an object of type HttpResponse, and all the lambda handler needs to do then is convert a standard response to APIGatewayProxyResponse. Then the web server front-end won't need to replicate this logic.